Off-Top-App/SettingsService
# OT-DeploySettingsMicroservice | Deployed settings microservice and database onto a Kubernetes cluster

**What I Did:**
1. Created several deployment scripts for the settings microservice as well as the database.
2. Ran these deployment scripts to deploy our microservice and database onto a Kubernetes cluster

**How To Test**
1. Follow this [**documentation**](https://docs.google.com/document/d/19_XEg7ZIM2NnJSQFBLUO1noKUih4YcKEMyRsV1iT0Is/edit) to set up a Kubernetes cluster through minikube
2. Follow this [**documentation**](https://docs.google.com/document/d/1cNMfzGiPlpmmegINQH_sEqL2pkWsgsMuvr8dOAFg4EM/edit) in order for you to deploy the settings microservice onto Kubernetes.
3. The documentation only has information on how to test the /ping endpoint, but in order for us to test if the database is also hosted, we will need to hit the POST request. Information on how to test that RESTful API can be found [**here**](https://docs.google.com/document/d/1cMh5mi0ZsyY7lW2JRwScSWB-uihMAF9s84shHdIIzMk/edit#bookmark=id.rq2wuh78pwve). 
   - **NOTE:** Make sure to use the URL provided when running `minikube service <service-name> --url` instead of `http://localhost:3000/` 
   - Make sure that you're able to get a response that looks like the one below. If so then you too have successfully deployed a microservice with its database onto Kubernetes.
```
{
    "success": 1,
    "data": {
        "app_color": "pink",
        "vibration_type": "medium",
        "default_categories": "[math, english, science]",
        "alert_type": "global",
        "user_email": "jdane@gmail.com"
    }
}
```
